File: app/src/main/java/com/gofd/mcauto/android_seed_project/MainActivity.java
package com.gofd.mcauto.android_seed_project;

import android.graphics.Color;
import android.graphics.drawable.Drawable;
import android.support.v4.app.Fragment;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.mikepenz.fontawesome_typeface_library.FontAwesome;
import com.mikepenz.iconics.IconicsDrawable;
import com.mikepenz.ionicons_typeface_library.Ionicons;

import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;
import devlight.io.library.ntb.NavigationTabBar;

public class MainActivity extends AppCompatActivity {
  @BindView(R.id.ntb_horizontal)
  NavigationTabBar navTab;
  private ViewPager viewPager;
  private MainTabPagerAdapter viewPagerAdapter;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate (savedInstanceState);
    setContentView (R.layout.activity_main);
    ButterKnife.bind (this);
    initUI ();
  }

  private void initUI() {
    viewPager = findViewById (R.id.vp_horizontal_ntb);
    viewPagerAdapter = new MainTabPagerAdapter (getSupportFragmentManager ());
    initPages ();
    viewPager.setAdapter (viewPagerAdapter);
    initNavigationBar ();
  }

  private void initPages() {
    viewPagerAdapter.addFragment (SampleFragment.newInstance ());
    viewPagerAdapter.addFragment (new Fragment ());
  }

  private void initNavigationBar() {

    int activeColor = getResources ().getColor (R.color.tab_active);
    int inactiveColor = getResources ().getColor (R.color.tab_inactive);

    navTab.setBgColor (Color.WHITE);
    navTab.setActiveColor (activeColor);
    navTab.setInactiveColor (inactiveColor);
    navTab.setIsTitled (true);

    final ArrayList<NavigationTabBar.Model> models = new ArrayList<> ();

    Drawable iList = new IconicsDrawable (this, Ionicons.Icon.ion_ios_folder_outline).sizeDp (24);
    Drawable iCog = new IconicsDrawable (this).icon (FontAwesome.Icon.faw_cog).sizeDp (24);
    String modelBgColor = "#ffffff";
    models.add (
            new NavigationTabBar.Model.Builder (
                    iList,
                    Color.parseColor (modelBgColor))
                    .title (getString (R.string.main_nav_first))
                    .build ()
    );
    models.add (
            new NavigationTabBar.Model.Builder (
                    iCog,
                    Color.parseColor (modelBgColor))
                    .title (getString (R.string.main_nav_second))
                    .build ()
    );


    navTab.setModels (models);
    navTab.setViewPager (viewPager, 0);

    navTab.setBehaviorEnabled (true);
    navTab.setBadgeSize (20);

    navTab.setOnTabBarSelectedIndexListener (new NavigationTabBar.OnTabBarSelectedIndexListener () {
      @Override
      public void onStartTabSelected(final NavigationTabBar.Model model, final int index) {

      }

      @Override
      public void onEndTabSelected(final NavigationTabBar.Model model, final int index) {

      }
    });

    navTab.setOnPageChangeListener (new ViewPager.OnPageChangeListener () {
      @Override
      public void onPageScrolled(final int position, final float positionOffset, final int positionOffsetPixels) {

      }

      @Override
      public void onPageSelected(final int position) {

      }

      @Override
      public void onPageScrollStateChanged(final int state) {

      }
    });
    navTab.bringToFront ();
  }
}

File: app/src/main/java/com/gofd/mcauto/android_seed_project/MainActivity.kt
package com.gofd.mcauto.android_seed_project

import android.graphics.Color
import android.graphics.drawable.Drawable
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity
import com.mikepenz.fontawesome_typeface_library.FontAwesome
import com.mikepenz.iconics.IconicsDrawable
import com.mikepenz.ionicons_typeface_library.Ionicons
import devlight.io.library.ntb.NavigationTabBar
import devlight.io.library.ntb.NavigationTabBar.Model.Builder
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
  private lateinit var viewPagerAdapter: MainTabPagerAdapter

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)
    initUI()
  }

  private fun initUI() {
    initPages()
    initNavigationBar()
  }

  private fun initPages() {
    viewPagerAdapter = MainTabPagerAdapter(supportFragmentManager)
    viewPagerAdapter.addFragment(SampleFragment().newInstance())
    viewPagerAdapter.addFragment(Fragment())
    viewPager.adapter = viewPagerAdapter
  }

  private fun initNavigationBar() {
    val activeColor = resources.getColor(R.color.tab_active)
    val inactiveColor = resources.getColor(R.color.tab_inactive)
    navTab.bgColor = Color.WHITE
    navTab.activeColor = activeColor
    navTab.inactiveColor = inactiveColor
    navTab.setIsTitled(true)

    navTab.models = initNavTabs()
    navTab.setViewPager(viewPager)
    navTab.isBehaviorEnabled = true
    navTab.badgeSize = 20f
    navTab.bringToFront()
  }

  private fun initNavTabs(): ArrayList<NavigationTabBar.Model> {
    val tabs: ArrayList<NavigationTabBar.Model> = ArrayList()
    val listIcon: Drawable = IconicsDrawable(this, Ionicons.Icon.ion_ios_folder_outline).sizeDp(24)
    val settingIcon: Drawable = IconicsDrawable(this).icon(FontAwesome.Icon.faw_cog).sizeDp(24)
    val listTitle = getString(R.string.main_nav_first)
    val cogTitle = getString(R.string.main_nav_second)
    val listTab = Builder(listIcon, Color.WHITE).title(listTitle).build()
    val settingTab = Builder(settingIcon, Color.WHITE).title(cogTitle).build()

    tabs.add(listTab)
    tabs.add(settingTab)

    return tabs
  }
}
